lib/azure/azHttpClient.js

```diff
@@ -94,35 +97,92 @@ class AzHttpClient {
                     content: method == 'PUT' ? policy : undefined
                 });
             });
-            let batchResponses = yield this.processBatchRequestSync(batchRequests);
+            let batchCallSize = (method == 'PUT') ? SYNC_BATCH_CALL_SIZE : ASYNC_BATCH_CALL_SIZE;
+            let batchResponses = yield this.processBatchRequestAsync(batchRequests, batchCallSize);
             // We need to return response in the order of request.
             batchResponses.sort(this.compareBatchResponse);
             return batchResponses;
         });
     }
-    processBatchRequestSync(batchRequests) {
+    processBatchRequestAsync(batchRequests, batchCallSize) {
         return __awaiter(this, void 0, void 0, function* () {
             let batchResponses = [];
+            let pendingRequests = [];
             if (batchRequests.length == 0) {
                 return Promise.resolve([]);
             }
-            // For sync implementation we will divide into chunks of 20 requests.
-            const batchRequestsChunks = utilities_1.splitArray(batchRequests, SYNC_BATCH_CALL_SIZE);
+            const batchRequestsChunks = utilities_1.splitArray(batchRequests, batchCallSize);
```

Comment:
pollPendingRequests can be merged with processBatchRequestAsync to have a single method. You can bring in the polling logic in here.